source/text.c

```diff
@@ -249,24 +283,242 @@ void C2D_DrawText(const C2D_Text* text, u32 flags, float x, float y, float z, fl
 	}
 	if (flags & C2D_WithColor)
 		color = va_arg(va, u32);
+	if (flags & C2D_WordWrap)
+		maxWidth = (float)va_arg(va, double); // Passed as float, but varargs promotes to double.
 
 	va_end(va);
 
 	C2Di_SetCircle(false);
 
-	for (cur = begin; cur != end; ++cur)
+	// Get information about lines
+	struct line {
+		u32 words;
+		u32 wordStart;
+	};
+	struct line lines[text->lines];
```

Comment:
Is this needed for *all* variants of C2D_DrawText? Shouldn't it be moved somewhere else?

Reply:
All of that information is necessary for wrapping, and a bunch of it is also necessary for justified text. I figured that calculating it twice (in the case of justified wrapped text) was worse than calculating it once for everything

Reply:
I mean, the original code didn't need it. Is the original mode still accessible, or was that just deleted? I think the original mode should still be accessible, and the new modes should be handled separately (probably?)

Reply:
Most of that code was actually a straight cut-paste from the original justified code. I suppose having a top-level if/else for wrapped and nonwrapped modes would be possible, but it would be a lot of code duplication

Reply:
But the fact that it's calculating a bunch of unnecessary stuff is really concerning, and it's especially undesirable for a hot function such as this one :\

Reply:
I can definitely change the scoping, but it will duplicate pretty much the entire switch, plus that code into the nonwrapped justified case. Would you rather have that duplication or the unconditional calculations? Your call.

Reply:
Aren't you able to break this up in smaller (static/static inline) functions, instead of repeating the code? I don't want unnecessary calculations to exist at runtime.

Reply:
That may be possible. I'll look into that after I fix the more trivial issues.